.cloudbuild/prombench.yaml

```diff
@@ -0,0 +1,490 @@
+        echo "true" > /workspace/EXIT_EARLY
+        exit 1
+      fi
+
+      # User Verification (https://docs.github.com/en/graphql/reference/enums#commentauthorassociation)
+      if ! ([ "${_COMMENTER_ROLE}" == "OWNER" ] || [ "${_COMMENTER_ROLE}" == "MEMBER" ] || [ "${_COMMENTER_ROLE}" == "COLLABORATOR" ])
+      then
+
+        GITHUB_ORG=$(echo "${_GIT_REPO}" | cut -d'/' -f1)
+        GITHUB_REPO=$(echo "${_GIT_REPO}" | cut -d'/' -f2)
+        export GITHUB_ORG="$${GITHUB_ORG}" && export GITHUB_REPO="$${GITHUB_REPO}"
+
+        COMMENT_MSG="${_COMMENTER} does not have admin or write privileges. Therefore cannot execute the benchmark command."
+        GITHUB_URL="https://api.github.com/repos/$${GITHUB_ORG}/$${GITHUB_REPO}/issues/${_PR_NUMBER}/comments"
+        echo "$${COMMENT_MSG}"
+        
+        # Commenting on the GitHub PR
+        curl --silent -L -X POST \
+          -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $${GH_PAT}" -H "X-GitHub-Api-Version: 2022-11-28" \
+          "$${GITHUB_URL}" \
+          -d '{"body": "'"$${COMMENT_MSG}"'"}'
+        
+        echo "true" > /workspace/EXIT_EARLY
+        exit 1
+      fi
+
+  - id: "Setting environment variables"
+    name: gcr.io/cloud-builders/gcloud
+    env:
+    - '_PR_COMMENT_TEXT=$_PR_COMMENT_TEXT'
+    allowExitCodes: [1]
+    entrypoint: "bash"
+    args:
+      - "-c"
+      - |
+        if [ -f "/workspace/EXIT_EARLY" ]
+        then
+          export EXIT_EARLY=$(cat /workspace/EXIT_EARLY)
+          if [ "$${EXIT_EARLY}" == "true" ]
+          then
+            exit 1
+          fi
+        fi
+
+        COLLECTOR_TAG_OR_CANCEL=$(echo "${_PR_COMMENT_TEXT}" | head -n 1 | awk '{ print $2 }')
+        if [ "$${COLLECTOR_TAG_OR_CANCEL}" == "cancel" ]
+        then          
+          echo "true" > /workspace/CANCEL_BENCHMARK
+        
+        else  
+          echo "GMP Collector Tag: $${COLLECTOR_TAG_OR_CANCEL}"
+
+          export GCLOUD_TAG="[]"
+          if [[ "$${COLLECTOR_TAG_OR_CANCEL}" != '' ]]
+          then
+            GCLOUD_TAG=$(gcloud container images list-tags --filter="tags:$${COLLECTOR_TAG_OR_CANCEL}" --format=json $${PROMETHEUS_IMAGE_REPOSITORY})
+          fi
+            
+          if ([[ "$${GCLOUD_TAG}" == "[]" ]] || [[ "$${COLLECTOR_TAG_OR_CANCEL}" == '' ]])
+          then
+            echo "Either the given tag is not valid or it is empty. The latest collector tag would be used."
+            DEFAULT_COLLECTOR_TAG=$(gcloud container images list-tags "$${PROMETHEUS_IMAGE_REPOSITORY}" --sort-by=UPDATE_TIME,TAGS | tail -1 | awk '{ print $2 }')
+            echo "$${DEFAULT_COLLECTOR_TAG}" > /workspace/PROMETHEUS_IMAGE_VERSION
+            export PROMETHEUS_IMAGE_VERSION="$${DEFAULT_COLLECTOR_TAG}"
+          else
+            echo "$${COLLECTOR_TAG_OR_CANCEL}" > /workspace/PROMETHEUS_IMAGE_VERSION
+            export PROMETHEUS_IMAGE_VERSION="$${COLLECTOR_TAG_OR_CANCEL}"
+          fi
+          
```

Comment:
101 and 104 can be on line 107 as echo "$${PROMETHEUS_IMAGE_VERSION}" > /workspace/PROMETHEUS_IMAGE_VERSION